src/tracer/models.ts

```diff
@@ -0,0 +1,32 @@
+import { Event } from '../client';
+import { Evaluation as TestingEvaluation } from '../testing';
+import { ArbitraryProperties, PromptTracking } from '../types';
+
+export type TracerEvent = Omit<Event, 'id' | 'traceId'> & { traceId?: string };
+export type Evaluation = TestingEvaluation;
+
+export interface SendEventArgs {
+  traceId?: string;
+  spanId?: string;
+  parentSpanId?: string;
+  timestamp?: string;
+  properties?: ArbitraryProperties;
+  promptTracking?: PromptTracking;
+  evaluators?: BaseEventEvaluator[];
+}
+
+export abstract class BaseEventEvaluator {
+  abstract get id(): string;
+
+  maxConcurrency: number = 10;
+
+  abstract evaluateEvent(args: {
+    event: TracerEvent;
+  }): Evaluation | Promise<Evaluation>;
+}
+
+export interface ClickHouseEvaluationObject extends Evaluation {
+  id: string;
+  metadata?: Record<string, string>;
+  evaluatorExternalId: string;
+}
```

Comment:
I think we should name this EvaluationEvent or EvaluationResult or something?

Reply:
Changed this to `EvaluationWithIds`